t/10_susedistribution.t

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/perl
+# Copyright @ SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+use testapi;
+use Test::MockModule;
+use Test::MockObject;
+
+subtest 'handle_password_prompt on serial terminal checks wait_serial' => sub {
+    set_var('VIRTIO_CONSOLE', '1');
+    set_var('BACKEND', 'qemu');
+    my $testapi_mocked = Test::MockModule->new('testapi');
+    $testapi_mocked->mock('is_serial_terminal' => sub { return 1 });
+    $testapi_mocked->mock('type_string' => sub { return 'randompass' });
+    $testapi_mocked->noop('send_key', 'assert_screen');
+    $testapi_mocked->mock('wait_serial' => sub { return 'Password:' });
+    require susedistribution;
```

Comment:
The truth is that I was provided with this solution as the `use susedistribution;` seemed to not working. @Tina explained this to me on /#discuss-perl/. 

>  with this code the mocking will happen first, and then susedistribution will get the mocked functions instead

and also suggested to mock `susedistribution`.
But the tests passed i didnt bother to study them further.

Reply:
I'm not `@tina` but @perlpunk.

Reply:
I suggested to use require in your example because it only had a single test.
I'm now rewriting the test with mocking susedistribution, which was my second suggestion.
please stand by